ddp-workspace/projects/ddp-sdk/src/lib/components/address/addressInput.service.ts

```diff
@@ -0,0 +1,414 @@
+import { ChangeDetectorRef, OnDestroy } from '@angular/core';
+import { CountryAddressInfo } from '../../models/countryAddressInfo';
+import { BehaviorSubject, merge, Observable, of, pipe, Subject, UnaryFunction, zip } from 'rxjs';
+import {
+  catchError,
+  concatMap,
+  delay,
+  distinctUntilChanged,
+  filter,
+  map,
+  mapTo,
+  pairwise,
+  pluck,
+  scan,
+  share,
+  shareReplay,
+  skip,
+  startWith,
+  switchMap,
+  take,
+  takeUntil,
+  tap,
+  withLatestFrom
+} from 'rxjs/operators';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { CountryService } from '../../services/addressCountry.service';
+import * as _ from 'underscore';
+import { Address } from '../../models/address';
+import { AddressService } from '../../services/address.service';
+
+
+type CountryCache = Record<string, CountryAddressInfo>;
+
+export interface AddressInputComponentState {
+  name: string;
+  country: string;
+  countryInfo?: CountryAddressInfo;
+  street1: string;
+  street2: string;
+  city: string;
+  zip: string;
+  phone: string;
+  guid: string;
+  isReadOnly: boolean;
+  formDataSource?: 'INPUT' | 'COMPONENT';
+  formData: any;
+}
+
+export class AddressInputService implements OnDestroy {
```

Comment:
Probably it would be better to move `AddressInputService` under `ddp-workspace\projects\ddp-sdk\src\lib\services`